### ISSUE TITLE
Persisting downloads not working

### ISSUE THREAD

**Issue by carlosperezm** [microsoft/playwright-python#705]
Hi,
I am trying to persist downloaded files by setting a folder called downloads with `downloads_path` as stated here: https://playwright.dev/python/docs/downloads/

Unfortunately, the files are deleted as soon as the session ends.

Here is a minimal example:

```python
import asyncio
from playwright.async_api import async_playwright

async def main():
    async with async_playwright() as p:
        browser = await p.chromium.launch(
            downloads_path="./downloads",
        )
        page = await browser.new_page(accept_downloads=True)
        await page.goto("https://unsplash.com/photos/tn57JI3CewI")

        async with page.expect_download() as download_info:
            # Perform the action that initiates download
            await page.click("._2vsJm")
        download = await download_info.value
        # Wait for the download process to complete
        path = await download.path()
        print(download.url, path)
        await page.close()
        await browser.close()

asyncio.run(main())
```

Any guidance is greatly appreciated.

Thanks in advance.

**Comment by mxschmitt** [comment]
Hi, this is currently expected, you can use the download.save_as method to store it in some other location so it's not getting deleted after the browser is closed: https://playwright.dev/python/docs/api/class-download#downloadsave_aspath

**Comment by carlosperezm** [comment]
Thank you for your reply.

Now I am trying with the following:

```diff
import asyncio
from playwright.async_api import async_playwright

async def main():
    async with async_playwright() as p:
        browser = await p.chromium.launch(
            downloads_path="./downloads",
        )
        page = await browser.new_page(accept_downloads=True)
        await page.goto("https://unsplash.com/photos/tn57JI3CewI")

        async with page.expect_download() as download_info:
            # Perform the action that initiates download
            await page.click("._2vsJm")
        download = await download_info.value
        # Wait for the download process to complete
        path = await download.path()
        print(download.url, path)
+       await download.save_as(""./downloads"") 

        await page.close()
        await browser.close()

asyncio.run(main())
```

Unfortunately it gives me this error
`PermissionError: [Errno 13] Permission denied: './downloads' `

What am I doing wrong here?

**Comment by mxschmitt** [comment]
You don't have permissions to write into the download folder. Try to specify an absolute path to a location where you have access, also saveAs accepts only a file, not a folder.

**Comment by carlosperezm** [comment]
Thanks, after giving the full path with the file name it worked.
Maybe could you add an example to the docs to avoid this confusion for anyone in the future?

**Comment by mxschmitt** [comment]
I created a PR #722 which should fix this issue for further users.

**Comment by mxschmitt** [comment]
Which version of Playwright are you using? It should be fixed already in v1.11.2

**Comment by mxschmitt** [comment]
Closed as part of the triage, please reopen / create a new issue if its still persistent.
